pkg/controllers/status/work_status_controller.go

```diff
@@ -400,6 +388,30 @@ func (c *WorkStatusController) reflectStatus(ctx context.Context, work *workv1al
 	})
 }
 
+func (c *WorkStatusController) interpretHealth(clusterObj *unstructured.Unstructured, work *workv1alpha1.Work) workv1alpha1.ResourceHealth {
+	// For kind that doesn't have health check, we treat it as healthy.
+	healthy := true
+	var err error
+	if c.ResourceInterpreter.HookEnabled(clusterObj.GroupVersionKind(), configv1alpha1.InterpreterOperationInterpretHealth) {
+		healthy, err = c.ResourceInterpreter.InterpretHealth(clusterObj)
+	} else {
+		klog.V(5).Infof("skipping health status of objects: %v %s/%s.", clusterObj.GroupVersionKind(), clusterObj.GetNamespace(), clusterObj.GetName())
+	}
+
+	var resourceHealth workv1alpha1.ResourceHealth
+	if err != nil {
+		resourceHealth = workv1alpha1.ResourceUnknown
+		c.EventRecorder.Eventf(work, corev1.EventTypeWarning, events.EventReasonInterpretHealthFailed, "Interpret health of object(%s/%s/%s) failed, err: %s.", clusterObj.GetKind(), clusterObj.GetNamespace(), clusterObj.GetName(), err.Error())
+	} else if healthy {
+		resourceHealth = workv1alpha1.ResourceHealthy
+		c.EventRecorder.Eventf(work, corev1.EventTypeNormal, events.EventReasonInterpretHealthSucceed, "Interpret health of object(%s/%s/%s) as healthy.", clusterObj.GetKind(), clusterObj.GetNamespace(), clusterObj.GetName())
```

Comment:
One question, do we still need to report the event in the skip situation?

Reply:
Good question, open to feedback. I personally would prefer if we didn't record the event

Reply:
cc @RainbowMango